relay-metrics/benches/benches.rs

```diff
@@ -1,158 +1,231 @@
+                num_buckets: 100_000,
+                metric_ids_generator: NumbersGenerator::new(1..100),
+                project_keys_generator: NumbersGenerator::new(1..100),
+                timestamp_shifts_generator: NumbersGenerator::new(10..50),
+            },
+        ),
+        (
+            "all non-backdated metrics",
+            BucketsGenerator {
+                percentage_backdated: 0.0,
+                num_buckets: 100_000,
+                metric_ids_generator: NumbersGenerator::new(1..100),
+                project_keys_generator: NumbersGenerator::new(1..100),
+                timestamp_shifts_generator: NumbersGenerator::new(10..50),
+            },
+        ),
+        (
+            "a lot of metrics from a lot of projects with mix of backdated buckets and non",
+            BucketsGenerator {
+                percentage_backdated: 0.5,
+                num_buckets: 1_000_000,
+                metric_ids_generator: NumbersGenerator::new(1..1_000_000),
+                project_keys_generator: NumbersGenerator::new(1..1_000_000),
+                timestamp_shifts_generator: NumbersGenerator::new(10..50),
+            },
+        ),
     ];
 
-    for input in &inputs {
-        c.bench_with_input(
-            BenchmarkId::new("bench_insert_metrics", input),
+    for (input_name, input) in &inputs {
+        let mut group = c.benchmark_group(*input_name);
+        group.throughput(criterion::Throughput::Elements(input.num_buckets as u64));
+        group.bench_with_input(
+            BenchmarkId::new("bench_insert_metrics", input_name),
             &input,
-            |b, &input| {
+            |b, input| {
                 b.iter_batched(
                     || {
+                        let timestamp = UnixTimestamp::now();
                         let aggregator: Aggregator = Aggregator::new(config.clone());
-                        (aggregator, input.get_buckets())
+                        (aggregator, input.get_buckets(timestamp))
                     },
                     |(mut aggregator, buckets)| {
                         for (project_key, bucket) in buckets {
-                            aggregator.merge(project_key, bucket, None).unwrap();
+                            #[allow(clippy::unit_arg)]
+                            black_box(
+                                aggregator
+                                    .merge(
+                                        black_box(project_key),
+                                        black_box(bucket),
+                                        black_box(None),
+                                    )
+                                    .unwrap(),
+                            );
                         }
                     },
                     BatchSize::SmallInput,
                 )
             },
         );
 
-        c.bench_with_input(
-            BenchmarkId::new("bench_flush_metrics", input),
+        group.bench_with_input(
```

Comment:
Shouldn't be the same group, this is a different benchmark with a different throughput! (no for loop in the body).